draft-ietf-quic-transport.md

```diff
@@ -7411,6 +7402,43 @@ The initial contents of this registry are shown in {{iana-error-table}}.
 
 --- back
 
+# Sample Packet Number Encoding Algorithm {#sample-packet-number-encoding}
+
+The pseudo-code in {{alg-encode-pn}} shows how an implementation can select
+an appropriate size for packet number encodings.
+
+The EncodePacketNumber function takes two arguments:
+
+* full_pn is the full packet number of the packet being sent.
+* largest_acked is the largest packet number which has been acknowledged by the
+  recipient in the current packet number space, if any
+
+~~~
+EncodePacketNumber(full_pn, largest_acked):
+  // If no packets in the current space have been
+  // acknowledged, the full packet number is used.
+  if largest_acked is None:
+    return full_pn
```

Comment:
Whoops, missed this.  You can safely assign num_acked to `full_pn + 1` in this case.  Returning early is bad for the same reasons.

Reply:
You can set min_bits to log2(full_pn) (without the +1) as well, I guess.

Reply:
@martinthomson : I don't follow what setting num_acked to `full_pn + 1` gets you. Why is this early return not good enough?

Reply:
The early return takes a number and returns it.  We are looking for an encoded value that includes a length (in bytes).  If full_pn is 10, returning that doesn't tell the caller how many bytes to write out.

Running this through the full encoding algorithm avoids that problem.